### PR TITLE
WIP: Improve psk handling

### LINKED ISSUE #96: pytradfri 4.0.0 not working
### Where are you using pytradfri (eg stand-alone, Home Assistant etc)
standalone

### Version of pytradfri
4.0.0

### Expected behaviour
working again with gateway firmware 1.2.42

### Actual behaviour
running this:

```
#!/usr/bin/env python3

import sys
from pytradfri import Gateway
from pytradfri.api.libcoap_api import APIFactory

api_factory = APIFactory(sys.argv[1])
with open('/mnt/rw/ikea/gateway_psk.txt', 'a+') as file:
    file.seek(0)
    psk = file.read()
    if psk:
        api_factory.psk = psk.strip()
    else:
        psk = api_factory.generate_psk(sys.argv[2])
        print('Generated PSK: ', psk)
        file.write(psk)
api = api_factory.request
```

`root@raspberrypi:~# /opt/testing 192.168.21.26 <devicekey>`

gives:
```
Traceback (most recent call last):
  File "/opt/testing", line 13, in <module>
    psk = api_factory.generate_psk(sys.argv[2])
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/api/libcoap_api.py", line 156, in generate_psk
    self._psk = self.request(command)
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/api/libcoap_api.py", line 93, in request
    return self._execute(api_commands)
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/api/libcoap_api.py", line 87, in _execute
    api_command.result = _process_output(return_value, parse_json)
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/command.py", line 71, in result
    self._result = self._process_result(value)
  File "/usr/local/lib/python3.5/dist-packages/pytradfri/gateway.py", line 28, in process_result
    return result[ATTR_PSK]
TypeError: 'NoneType' object is not subscriptable
```
???

### PR DESCRIPTION
**Pull request by ggravlingen**
This closes #96 given that you have either aiocoap or libcoap installed.

### CODE REVIEW

File: example_async.py [hunk 1]
```diff
@@ -35,16 +49,26 @@
...
-    with open('gateway_psk.txt', 'a+') as file:
-        file.seek(0)
-        psk = file.read()
-        if psk:
-            api_factory.psk = psk.strip()
-        else:
-            psk = yield from api_factory.generate_psk(sys.argv[2])
+    conf = load_json(CONFIG_FILE)
+
+    try:
```
Comment (ggravlingen) [inline example_async.py]: @lwis these lines (or something along the line of these lines) will appear in all examples. Should I extract this to a class instead?

Comment (lwis) [reply example_async.py]: Hmm, we risk making the examples too complex if we make reference to other files. And this doesn't really belong in the main lib. Not sure!

Comment (ggravlingen) [reply example_async.py]: Agree and was unsure myself, hence reaching out. Let's keep it as is for now.

File: pytradfri/api/libcoap_api.py [hunk 2]
```diff
@@ -149,7 +149,6 @@ def generate_psk(self, security_key):
         if not self._psk:
             existing_psk_id = self._psk_id
 
-            self._psk_id = 'Client_identity'
```
Comment (ggravlingen) [inline pytradfri/api/libcoap_api.py]: @lwis I believe this is the reason users may have experienced problems. I got this back from the gateway when debugging. We would expect 9090 to be pytradfri in current version and uuid.uuid4().hex in proposed update of the lib.
```
{'9090': 'Client_identity'}
['coap-client', '-u', 'Client_identity', '-k', 'PnHhjOjepj8vhbZB', '-v', '0', '-m', 'post', '-f', '-', 'coaps://192.168.0.129:5684/15011/9063']
```

Comment (ggravlingen) [reply pytradfri/api/libcoap_api.py]: Clarification: the output above is from before I removed the line.

Comment (ggravlingen) [reply pytradfri/api/libcoap_api.py]: You are right in that Client_identity user generates psk but we also expect 9090 to contain the identity we want a psk for: https://twitter.com/home_assistant?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Eauthor

It works as expected in the aiocoap api but not in the libcoap api.

Comment (lwis) [reply pytradfri/api/libcoap_api.py]: Is the request method broken?

Comment (ggravlingen) [reply pytradfri/api/libcoap_api.py]: I'm unsure, libcoap seems to work fine if the key has already been generated and is available in the file (json or in the gateway_psk.txt file.)

**Pull request merged**